lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -354,7 +354,7 @@ public function get_items_permissions_check( $request ) {
 			$post = get_post( (int) $request['post'] );
 
 			if ( $post && ! $this->check_read_post_permission( $post ) ) {
-				return new WP_Error( 'rest_cannot_read_post', __( 'Sorry, you cannot read the post for this comment.' ) );
+				return new WP_Error( 'rest_cannot_read_post', __( 'Sorry, you cannot read the post for this comment.' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
@danielbachhuber Would a `403` status code be more accurate here?  Even if a user was logged-in they may not have permission to read private posts.